vllm/spec_decode/target_model_runner.py

```diff
@@ -0,0 +1,66 @@
+from typing import List, Optional
+
+from vllm.config import (CacheConfig, DeviceConfig, LoadConfig, LoRAConfig,
+                         ModelConfig, MultiModalConfig, ParallelConfig,
+                         PromptAdapterConfig, SchedulerConfig)
+from vllm.logger import init_logger
+from vllm.sequence import SequenceGroupMetadata
+from vllm.worker.model_runner import (ModelInputForGPUWithSamplingMetadata,
+                                      ModelRunner)
+
+logger = init_logger(__name__)
+
+
+class TargetModelRunner(ModelRunner):
+    """Specialized model runner for speculative decoding target model.
```

Comment:
nit: add comment explaining why we do this
```
In speculative decoding, the logprobs selected may not be the same ones as
selected by the target model sampling. This means that the time spent in the
logprob calculation of the target model is time wasted, since we calculate
logprobs after deciding which tokens are accepted. For this reason we disable
logprobs in the target model so scoring is faster.
```

Reply:
Done.